STYLE_GUIDE.md

```diff
@@ -0,0 +1,481 @@
+
+## Markdown Formatting
+
+### Links
+**Use inline markdown links** for all references:
+```markdown
+[link text](https://example.com)
+```
+
+**Avoid** reference-style links unless you're using the same URL multiple times in the same post.
+
+### Emphasis
+
+#### Italics
+Use **single underscores** for italic text:
+```markdown
+_This is italic text_
+```
+
+**Common uses:**
+- Book, movie, or product names
+- Introducing new terms
+- Attribution at the end of posts: `_Thanks to Person Name for feedback._`
+- Foreign words or phrases
+- Metadata/preamble text: `_Discussed on [Hacker News](...)_`
+
+#### Bold
+Use **double asterisks** for bold text:
+```markdown
+**This is bold text**
+```
+
+**Common uses:**
+- Emphasizing key points within sentences
+- Important concepts or warnings
+- Section of text you want to stand out
+
+#### Code
+Use **backticks** for inline code:
+```markdown
+`someFunction()` or `variableName`
+```
+
+### Code Blocks
+
+Always specify the language for syntax highlighting:
+
+````markdown
+```javascript
+const foo = "bar";
+```
+
+```typescript
+type User = { name: string };
+```
+
+```python
+def hello():
+    return "world"
+```
+````
+
+**Common language identifiers:**
+- `javascript` (preferred over `js`)
+- `typescript` (preferred over `ts`)
+- `jsx`
+- `tsx`
+- `python`
+- `graphql`
+- `bash` or `shell`
+
+### Headers
+
+Use `##` (H2) for main section headers within posts. The post title is H1 and should not be repeated in the body.
+
+**Use sentence case** for headers, same as titles:
+```markdown
+## This is a section header
+## Making it fast and cheap
+## What's the difference?
+```
+
+**Avoid:**
+- Using `#` (H1) in post body
+- Title case for headers: ~~## This Is A Section Header~~
+
+### Horizontal Rules
+
+Use **three hyphens** for horizontal rules:
+```markdown
+---
+```
+
+**Common uses:**
+- Separating preamble/metadata from main content
+- Before attribution/acknowledgments at the end of posts
+- Between major conceptual sections (use sparingly)
+
+### Lists
+
+#### Unordered Lists
+Use **hyphens** for bullet points:
+```markdown
+- First item
+- Second item
+- Third item
+```
+
+#### Ordered Lists
+Use **numbers with periods**:
+```markdown
+1. First step
+2. Second step
+3. Third step
+```
+
+### Block Quotes
+
+Use `>` for block quotes:
+```markdown
+> This is a quoted passage.
+> It can span multiple lines.
+```
+
+## Post Structure
+
+### Preamble (Optional)
+Posts may optionally include italicized metadata at the beginning:
+
+```markdown
+_Discussed on [Hacker News](https://news.ycombinator.com/...)._
+
+---
+```
+
+Or:
+
+```markdown
+_TL;DR: Brief summary of the entire post._
+
+---
+```
+
+Always follow preamble with `---` to separate it from the main content.
+
+### Opening
+Start posts directly with the topic. No need for "Hello readers" or similar pleasantries.
+
+### Attribution and Thanks (Optional)
+When thanking contributors or providing attribution, place at the end after `---`:
```

Comment:
Clarify that the attribution should ideally include a link to that user's canonical online identity. A personal homepage if possible. If not, a social media or github profile.